eth/stagedsync/stage_interhashes.go

```diff
@@ -101,7 +101,11 @@ func SpawnIntermediateHashesStage(s *StageState, u Unwinder, tx kv.RwTx, cfg Tri
 				return trie.EmptyRoot, fmt.Errorf("Wrong trie root")
 			}
 			if cfg.hd != nil {
-				header := rawdb.ReadHeader(tx, headerHash, to)
+				header, err := cfg.blockReader.HeaderByHash(ctx, tx, headerHash)
+				if err != nil {
+					return trie.EmptyRoot, err
+				}
+
```

Comment:
need to check header == nil anyway

Reply:
if header is nil I believe we should return an error and force snapshots.verify since it means we have a gab